src/Microsoft.AspNetCore.Mvc.Razor/Internal/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -105,14 +108,52 @@ private TagHelperDescriptor CreateDescriptor(ViewComponentDescriptor viewCompone
                 descriptor.IsIndexer = false;
 
                 attributeDescriptors.Add(descriptor);
+
+                var indexerDescriptor = GetIndexerAttributeDescriptor(parameter, lowerKebabName);
+                if (indexerDescriptor != null)
+                {
+                    indexerDescriptors.Add(indexerDescriptor);
+                }
```

Comment:
Add a comment stating why we don't add required attribute descriptors for indexers